src/main/java/duke/Task.java

```diff
@@ -0,0 +1,78 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Task {
+    public enum TaskType {
+        TODO,
+        DEADLINE,
+        EVENT
+    }
+    protected String description;
+    protected boolean isDone;
+    protected TaskType taskType;
+
+    public Task(String description, TaskType taskType) {
+        this.description = description;
+        this.isDone = false;
+        this.taskType = taskType;
+
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    //function to mark as done
+    public void markAsDone() {
+        this.isDone = true;
+    }
+    //function to unmark
+    public void unMark(){
+        this.isDone = false;
+    }
+    public String toString(){
+        return "["+ this.getStatusIcon() +"] " + this.description;
+    }
+
+
+    public String toFileString(){
+        return String.format("%s |  %d | %s", getType(), isDone? 1:0, description);
+    }
+
+    public static Task fromFileString(String fileString){
+        String[] elements = fileString.split("\\s*\\|\\s*");
+        Boolean done = Integer.parseInt(elements[1]) == 1;
+        //for each type of task
```

Comment:
```suggestion
        // For each type of task
```

Comments should be capitalized and should have a space.